lucille-core/src/main/java/com/kmwllc/lucille/core/Document.java

```diff
@@ -627,7 +627,7 @@ default JsonNode getNestedJson(String[] pathSegments) {
    * @param name the nested field path to set the JsonNode at
    * @param value the JsonNode to set at the nested path
    */
-  default void setNestedJson(String name, JsonNode value) {
+  default void setNestedJson(String name, JsonNode value) throws IllegalArgumentException {
```

Comment:
IllegalArgumentException is an unchecked exception so you don't need to declare it
you still can declare it to be explicit, but I think it's more common not to declare